fuzz/fuzz_targets/cranelift-fuzzgen.rs

```diff
@@ -148,6 +149,8 @@ pub struct TestCase {
     /// Generate multiple test inputs for each test case.
     /// This allows us to get more coverage per compilation, which may be somewhat expensive.
     pub inputs: Vec<TestCaseInput>,
+    /// Should this `TestCase` be tested after optimizations.
+    pub should_optimize: bool,
 }
 
 impl fmt::Debug for TestCase {
```

Comment:
I can't highlight the actual line, but can we print a comment to the resulting test case if we choose the `should_optimize` flag? Or even better, add a different variant of `PrintableTestCase` that gets us a better representation of this test case?

When the fuzzer finds an issue we try to print a .clif file that reproduces the issue, that isn't always possible, but we try to get it as close as we can so that its easier to reproduce later.

In this case we don't have any way of running the interpreter after optimizations in the test case, but adding a comment would be really helpful so that we know that this is sort of a special case.

Reply:
I'd definitely like to have the optimized test case in the `Debug` output, but we have to be a little careful. If anything in the optimizer panics, we wouldn't get any debug output at all.

To keep things simple for the moment, I think I just want your first suggestion: add a comment to the `Debug` output indicating whether the test case compared the interpreter against the host, or against interpreting optimized CLIF.

I think we should add a command line mode for `cranelift-tools` that runs the optimization pass and pretty-prints the resulting CLIF. Then we don't need to do anything further in fuzzgen's `Debug` formatter.

Reply:
The possibly-overkill end of the spectrum is make some kind of `diff` utility. But that is out of scope for this PR (... probably). Should I just add a println before running the `PrintableTestCase`?

Reply:
Yeah, I think `write!`ing something before printing the test case sounds like a good idea.

Reply:
Thing is, `PrintableTestCase` prints some stuff before the test attributes

Reply:
No worries, just print out a CLIF comment indicating which way we went with the `compare_against_host` flag. Just so we know if we need a bit of manual work to reproduce the case. As long as it's in a comment format it shouldn't really affect the reproducibility of the test case.